zk/stages/stage_sequence_execute_utils.go

```diff
@@ -498,3 +499,69 @@ func (bdc *BlockDataChecker) AddTransactionData(txL2Data []byte) bool {
 
 	return false
 }
+
+type ResequenceBatchJob struct {
+	batchToProcess  []*dsTypes.FullL2Block
+	StartBlockIndex int
+	StartTxIndex    int
+	txIndexMap      map[common.Hash][]int
+}
+
+func NewResequenceBatchJob(batch []*dsTypes.FullL2Block) *ResequenceBatchJob {
+	return &ResequenceBatchJob{
+		batchToProcess:  batch,
+		StartBlockIndex: 0,
+		StartTxIndex:    0,
+		txIndexMap:      make(map[common.Hash][]int),
+	}
+}
+
+func (r *ResequenceBatchJob) HasMoreBlockToProcess() bool {
+	return r.StartBlockIndex < len(r.batchToProcess)
+}
+
+func (r *ResequenceBatchJob) CurrentBlock() *dsTypes.FullL2Block {
+	if r.HasMoreBlockToProcess() {
+		return r.batchToProcess[r.StartBlockIndex]
+	}
+	return nil
+}
+
+func (r *ResequenceBatchJob) YieldNextBlockTransactions(decoder zktx.TxDecoder) ([]types.Transaction, error) {
+	blockTransactions := make([]types.Transaction, 0)
+	if r.HasMoreBlockToProcess() {
+		block := r.CurrentBlock()
+		r.txIndexMap[block.L2Blockhash] = []int{r.StartBlockIndex, 0}
+
+		for i := r.StartTxIndex; i < len(block.L2Txs); i++ {
+			transaction := block.L2Txs[i]
+			tx, _, err := decoder(transaction.Encoded, transaction.EffectiveGasPricePercentage, block.ForkId)
+			r.txIndexMap[tx.Hash()] = []int{r.StartBlockIndex, i}
+			if err != nil {
+				return nil, fmt.Errorf("decode tx error: %v", err)
+			}
```

Comment:
Swap these two lines?
```suggestion
            if err != nil {
				return nil, fmt.Errorf("decode tx error: %v", err)
			}
			r.txIndexMap[tx.Hash()] = []int{r.StartBlockIndex, i}
```